vars/commonPipelineEnvironment.groovy

```diff
@@ -187,11 +187,23 @@ class commonPipelineEnvironment implements Serializable {
         [filename: '.pipeline/commonPipelineEnvironment/git/commitId', property: 'gitCommitId'],
         [filename: '.pipeline/commonPipelineEnvironment/git/commitMessage', property: 'gitCommitMessage'],
         [filename: '.pipeline/commonPipelineEnvironment/mtarFilePath', property: 'mtarFilePath'],
+        [filename: '.pipeline/commonPipelineEnvironment/configurationFile', property: 'configurationFile'],
         [filename: '.pipeline/commonPipelineEnvironment/abap/repositoryNames', property: 'abapRepositoryNames'],
     ]
 
     void writeToDisk(script) {
 
+        // The default config files were extracted from merged library resources
+        // by setupCommonPipelineEnvironment.groovy into folder .pipeline/tmp/defaults/.
+        List customDefaults = getCustomDefaults()
+        List customDefaultFiles = []
+        customDefaults.each { customDefault ->
+            def filepath =  ".pipeline/tmp/defaults/${customDefault}"
+            script.writeFile(file: filepath, text: script.libraryResource(customDefault))
+            customDefaultFiles.add(filepath)
+        }
+        script.writeFile file: '.pipeline/commonPipelineEnvironment/customDefaults', text: groovy.json.JsonOutput.toJson(customDefaultFiles)
```

Comment:
I am confused about whether downloaded defaults are no longer handled, but should be. Judging by the code where this came from, getCustomDefaults() doesn't only contain library resources, but also files passed as parameter (which may exist in the project) or have been downloaded on demand.

Reply:
valid point, I guess the non-Jenkins resources need to be handled properly.
I don't know the use-case here in details thus I might need some assistance.

Reply:
To be honest, I don't yet understand the motivation for this PR. All default files are already placed into ./pipeline by setupCommonPipelineEnvironment, stashed, and unstashed there as part of piperExecuteBin. If you need them in a sub-folder within ./pipeline, it would be easy to adjust setupCommonPipelineEnvironment accordingly.